pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -942,3 +979,106 @@ func (c *K8sOrchestrator) ClearFakeAttached(ctx context.Context, volumeID string
 	}
 	return nil
 }
+
+// initVolumeIDToNodesMap performs all the operations required to initialize
+// the volume id to node names map. It also watches for volume attachment add,
+// update & delete operations, and updates the map accordingly.
+func initVolumeIDToNodesMap(ctx context.Context) {
+	log := logger.GetLogger(ctx)
+	log.Debugf("Initializing volumeID to node name map")
+	k8sOrchestratorInstance.volumeIDToNodesMap = &volumeIDToNodesMap{
+		RWMutex: &sync.RWMutex{},
+		items:   make(map[string][]string),
+	}
+
+	// Set up kubernetes resource listener to listen events on volume attachments
+	k8sOrchestratorInstance.informerManager.AddVolumeAttachmentListener(
+		func(obj interface{}) { // Add.
+			volumeAttachmentAdded(obj)
+		},
+		func(oldObj interface{}, newObj interface{}) { // Update.
+			volumeAttachmentUpdated(oldObj, newObj)
+		},
+		func(obj interface{}) { // Delete.
+			volumeAttachmentDeleted(obj)
+		})
+}
+
+// volumeAttachmentAdded adds a new entry or updates an existing entry
+// in the volumeIDToNodeNames map if the volume attachment status is
+// true
+func volumeAttachmentAdded(obj interface{}) {
+	log := logger.GetLogger(context.Background())
+	volAttach, ok := obj.(*storagev1.VolumeAttachment)
+	if volAttach == nil || !ok {
+		log.Warnf("volumeAttachmentAdd: unrecognized object %+v", obj)
+		return
+	}
+
+	log.Debugf("volume attachment add: volume=%v", volAttach)
```

Comment:
Use `volumeAttachmentAdded` everywhere in this method

Reply:
Done.